source/common/formatter/substitution_formatter.cc

```diff
@@ -1645,5 +1659,34 @@ ProtobufWkt::Value SystemTimeFormatter::formatValue(
       format(request_headers, response_headers, response_trailers, stream_info, local_reply_body));
 }
 
+EnvironmentFormatter::EnvironmentFormatter(const std::string& key,
+                                           absl::optional<size_t> max_length) {
+  ASSERT(!key.empty());
+
+  const char* env_value = std::getenv(key.c_str());
+  if (env_value != nullptr) {
+    std::string env_string = env_value;
+    truncate(env_string, max_length);
```

Comment:
What is the scenario for truncate an env?

Reply:
It an common feature in the log formmater to avoid too long data output and is optional. If no valid max length is provided, then `truncate` do nothing.